File: src/Web/Medicine.Web.UseCases/Dto/DosingFrequencyReminderDto.cs
using Medicine.Entities.Models;
using Medicine.Entities.Models.Base.Interfaces;
using Medicine.Entities.Models.Translated;
using Medicine.Web.UseCases.DataLoaders.DataLoader;

namespace Medicine.Web.UseCases.Responses
{
    public class DosingFrequencyReminderDto : TranslationEntity<TranslatedDosingFrequencyReminder>
    {
        public double Quantity { get; set; }
        public string UsingDescription { get; set; }

        public int ReminderId { get; set; }

        public int DosingFrequencyId { get; set; }

        public async Task<DosingFrequencyDto>
            DosingFrequency(IResponseLoader<int, DosingFrequency, DosingFrequencyDto> dataLoader, CancellationToken ct)
        {
            var dosingFrequency = await dataLoader.LoadByCondition(x => x.Id == DosingFrequencyId);
            return dosingFrequency?.FirstOrDefault();
        }


        public async Task<IEnumerable<DosageLogDto>>
            DosageLogs(IResponseLoader<int, DosageLog, DosageLogDto> dataLoader, CancellationToken ct)
        {
            var dosageLogs = await dataLoader.LoadByCondition(x => x.DosageRecommendationId == Id);
            return dosageLogs;
        }

    }
}

File: front/medicine/src/app/shared/models/entity/DosingFrequencyReminderDto.ts
import { User } from "./User";
import { TranslatedDosingFrequencyReminder } from "./translate/TranslatedDosingFrequencyReminder";
import { DosageLogDto } from "./DosageLogDto";
import { DosingFrequencyDto } from "./DosingFrequencyDto";

export class DosingFrequencyReminderDto {
  dosingFrequency: DosingFrequencyDto = new DosingFrequencyDto();
  dosageLogs: DosageLogDto[] = [];
  quantity!: number;
  usingDescription!: string;
  reminderId!: number;
  dosingFrequencyId!: number;
  translations: TranslatedDosingFrequencyReminder[] = [];
  user?: User;
  userId?: number;
  id!: number;
  createdAt?: Date;
}
